engine/src/main/java/org/pentaho/di/kitchen/KitchenCommandExecutor.java

```diff
@@ -0,0 +1,414 @@
+
+    // Load a job
+    logDebug(  "Kitchen.Log.LoadingJobInfo" );
+    blockAndThrow( getKettleInit() );
+    JobMeta jobMeta = repository.loadJob( jobName, directory, null, null ); // reads last version
+    logDebug(  "Kitchen.Log.AllocateJob" );
+
+    return new Job( repository, jobMeta );
+  }
+
+  public Job loadJobFromFilesystem( String initialDir, String filename, Serializable base64Zip ) throws Exception {
+
+    if ( Utils.isEmpty( filename ) ) {
+      System.out.println( BaseMessages.getString( getPkgClazz(), "Kitchen.Error.canNotLoadJob" ) );
+      return null;
+    }
+
+    File zip;
+    if ( base64Zip != null && ( zip = decodeBase64ToZipFile( base64Zip, true ) ) != null ) {
+      // update filename to a meaningful, 'ETL-file-within-zip' syntax
+      filename = "zip:file:" + File.separator + File.separator + zip.getAbsolutePath() + "!" + filename;
+    }
+
+    blockAndThrow( getKettleInit() );
+    String fileName = filename;
+    // If the filename starts with scheme like zip:, then isAbsolute() will return false even though
+    // the path following the zip is absolute path. Check for isAbsolute only if the fileName does not start with scheme
+    if ( !KettleVFS.startsWithScheme( fileName ) && !FileUtil.isFullyQualified( fileName ) ) {
+      fileName = initialDir + fileName;
+    }
+
+    JobMeta jobMeta = new JobMeta( fileName, null, null );
+    return new Job( null, jobMeta );
+  }
+
+  protected void printJobParameters( Job job ) throws UnknownParamException {
+
+    if ( job != null && job.listParameters() != null ) {
+
+      for ( String pName : job.listParameters() ) {
+        printParameter( pName, job.getParameterValue( pName ), job.getParameterDefault( pName ), job.getParameterDescription( pName ) );
+      }
+    }
+  }
+
+  protected void printRepositoryStoredJobs( Repository repository, RepositoryDirectoryInterface directory ) throws KettleException {
+
+    logDebug( "Kitchen.Log.GettingLostJobsInDirectory", "" + directory );
+
+    String[] jobnames = repository.getJobNames( directory.getObjectId(), false );
+    for ( int i = 0; i < jobnames.length; i++ ) {
+      System.out.println( jobnames[i] );
+    }
+  }
+
+  protected void printRepositories( RepositoriesMeta repositoriesMeta ) {
+
+    if ( repositoriesMeta != null ) {
+
+      System.out.println( BaseMessages.getString( getPkgClazz(), "Kitchen.Log.ListRep" ) );
+
+      for ( int i = 0; i < repositoriesMeta.nrRepositories(); i++ ) {
+        RepositoryMeta rinfo = repositoriesMeta.getRepository( i );
+        System.out.println( "#" + ( i + 1 ) + " : " + rinfo.getName() + " [" + rinfo.getDescription() + "]  id=" + rinfo.getId() );
+      }
+    }
+  }
+
+  private <T extends Throwable> void blockAndThrow( Future<T> future ) throws T {
+
+    if ( future == null ) {
+      return;
+    }
+
+    try {
+      T e = future.get();
+      if ( e != null ) {
+        throw e;
+      }
+    } catch ( InterruptedException | ExecutionException e  ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)